openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIComponentMapper.java

```diff
@@ -863,6 +863,23 @@ private boolean isConstraintAnnotation(AnnotationNode annotation) {
         return false;
     }
 
+    /*
+     * This util is used to check whether an annotation is a constraint:Date annotation.
+     * Currently, we don't have support for mapping Date constraints to OAS hence we skip them.
+     * {@link <a href="https://github.com/ballerina-platform/ballerina-standard-library/issues/5049">...</a>}
+     * Once the above improvement is completed this method should be removed!
+     */
+    private boolean isNotConstraintDate(AnnotationNode annotation) {
```

Comment:
This method does two things:
1. Check whether the current constraint is a `Date` constraint.
2. If so update the compilation context with a warning. 

This breaks SRP. Lets check only whether the current constraint is a `Date` constraint or not in this method.